report/src/index.ts

```diff
@@ -45,21 +45,30 @@ export class ReportTabs extends DockPanel {
         this._wu.fetchResults().then(results => {
             if (this._results.length !== results.length) {
                 this._results = results;
-                results.filter(isProfileResult).forEach((result, i) => {
-                    let prevWidget;
+                const resultsArr = results.filter(isProfileResult);
+                let resultCount = 0;
+                const resultLimit = resultsArr.length;
+                const reportWidgetMap = {};
+                const reportNameMap = {};
+                resultsArr.forEach((result, i) => {
                     result.fetchRows()
                         .then(rows => {
-                            const r = new Report(rows);
+                            resultCount++;
+                            reportWidgetMap[i] = new Report(rows);
                             let tabName = 'Report ' + i;
                             if (result.Name) {
                                 tabName = result.Name;
                             }
-                            if (i === 0) {
-                                this.addWidget(r, tabName);
-                            } else {
-                                this.addWidget(r, tabName, "tab-after", prevWidget);
+                            reportNameMap[i] = tabName;
+                            if(resultCount === resultLimit){
```

Comment:
Should really use a `Promise.all` (code will get simpler):  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Your code will end up looking a but like:
```
Promise.all(resultsArr,map(r=>r.fetchRows())).then(results=> ...));
```

Reply:
@jbrundage Bump

Reply:
Ah okay, I'll make this change and test it today.

Reply:
@GordonSmith I've made the requested change:
https://github.com/hpcc-systems/DataPatterns/blob/4a35ecb22f05b332528136d12c1f3896044a072c/report/src/index.ts#L52-L68